src/Infrastructure.EntityFramework/Repositories/CipherRepository.cs

```diff
@@ -485,7 +495,7 @@ private async Task<DateTime> ToggleCipherStates(IEnumerable<Guid> ids, Guid user
             var query = from ucd in await (userCipherDetailsQuery.Run(dbContext)).ToListAsync()
                         join c in cipherEntitiesToCheck
                             on ucd.Id equals c.Id
-                        where ucd.Edit && ucd.DeletedDate == null
+                        where ucd.Edit && FilterDeletedDate(action, ucd)
```

Comment:
If this is working for you I have a feeling this is being evaluated client side instead of being transformed into something the database does. That's fine but in the long run we should try to work it in a way where we have the DB do this work.

Reply:
Discussed this internally, it's already client side so this is fine but one day it'd be great to make this all in the DB.